cmd/autofix_test.go

```diff
@@ -216,422 +215,597 @@ func TestSingleUnsupportedResource(t *testing.T) {
 }
 
 var testData = []struct {
-	orig   yaml.MapSlice
-	fixed  yaml.MapSlice
-	merged yaml.MapSlice
+	orig   *yaml.Node
+	fixed  *yaml.Node
+	merged *yaml.Node
 }{
 	// update
 	{
-		yaml.MapSlice{{Key: "k", Value: "v", Comment: "Comment"}},
-		yaml.MapSlice{{Key: "k", Value: "v2"}},
-		yaml.MapSlice{{Key: "k", Value: "v2", Comment: "Comment"}},
+		&yaml.Node{Kind: yaml.MappingNode, Tag: mapTag, Content: []*yaml.Node{
+			{Kind: yaml.ScalarNode, Tag: strTag, Value: "K", HeadComment: "Hi"},
+			{Kind: yaml.ScalarNode, Tag: strTag, Value: "V", LineComment: "Bye"},
+		}},
+		&yaml.Node{Kind: yaml.MappingNode, Tag: mapTag, Content: []*yaml.Node{
+			{Kind: yaml.ScalarNode, Tag: strTag, Value: "K"},
+			{Kind: yaml.ScalarNode, Tag: strTag, Value: "V2"},
+		}},
+		&yaml.Node{Kind: yaml.MappingNode, Tag: mapTag, Content: []*yaml.Node{
+			{Kind: yaml.ScalarNode, Tag: strTag, Value: "K", HeadComment: "Hi"},
+			{Kind: yaml.ScalarNode, Tag: strTag, Value: "V2", LineComment: "Bye"},
+		}},
```

Comment:
nit: because this is kind of a big PR, I would recommend trying to keep as many things constant as possible to make it easy for reviewers. In this test case both the structure and the content of the nodes change, which makes things a little more complicated - especially because I'm new to the codebase. If we want to test both comment types and we didn't support them before it might make sense to split out multiple smaller commits that reviewers can consider individually like: "move to yaml.v3 nodes", "add HeadComment to test cases"? Just a suggestion for future PRs.

Reply:
Yes, that makes sense! We did support them before, they just looked different. In the old format, aka what the fork uses, it would have looked like this:
```
yaml.MapSlice{
	{Comment: "Hi"},
	{Key: "K", Value: "V2", Comment: "Bye"},
},
```
but I agree that I should have done a direct conversion and kept the test contents the same. I'll keep it in mind for next time!